server/src/internalClusterTest/java/org/elasticsearch/indices/state/CloseIndexIT.java

```diff
@@ -486,19 +488,42 @@ public void testCommitIdInSearcher() throws Exception {
         assertAcked(client().admin().indices().prepareClose(indexName));
         assertIndexIsClosed(indexName);
         ensureGreen(indexName);
-        final String nodeWithPrimary = Iterables.get(internalCluster().nodesInclude(indexName), 0);
-        IndexShard shard = internalCluster().getInstance(IndicesService.class, nodeWithPrimary)
-            .indexService(resolveIndex(indexName)).getShard(0);
-        final String commitId;
-        try (Engine.SearcherSupplier searcherSupplier = shard.acquireSearcherSupplier(randomFrom(Engine.SearcherScope.values()))) {
-            assertNotNull(searcherSupplier.getCommitId());
-            commitId = searcherSupplier.getCommitId();
+        if (randomBoolean()) {
+            assertAcked(client().admin().indices().prepareUpdateSettings(indexName)
+                .setSettings(Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)));
+            internalCluster().ensureAtLeastNumDataNodes(2);
+            ensureGreen(indexName);
+        }
+        String[] searcherIds = new String[numberOfShards];
+        Set<String> allocatedNodes = internalCluster().nodesInclude(indexName);
+        for (String node : allocatedNodes) {
+            IndexService indexService = internalCluster().getInstance(IndicesService.class, node).indexServiceSafe(resolveIndex(indexName));
+            for (IndexShard shard : indexService) {
+                try (Engine.SearcherSupplier searcher = shard.acquireSearcherSupplier()) {
+                    assertNotNull(searcher.getCommitId());
```

Comment:
I wonder if we should rename the method to `getSearcherId()`, just like you did for the test case? Since it is no longer a commit-id, it might be a confusing name.

Reply:
++, I adjusted it in https://github.com/elastic/elasticsearch/pull/66668/commits/d2f78544003a1d9c5e56a5148b96b5d60bc1f0ff